core/src/main/java/com/datastax/oss/driver/internal/core/resolver/MockResolver.java

```diff
@@ -0,0 +1,63 @@
+package com.datastax.oss.driver.internal.core.resolver;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MockResolver implements Resolver {
```

Comment:
Take into account that it should be thread safe and you should be able to throw an exception.

Reply:
Also, since all instances of `MockResolver` should return same responses, you need to have `MockResolverFactory`.